Core/Python/add_device_to_static_group.py

```diff
@@ -57,32 +182,56 @@ def add_device_to_static_group(ome_ip_address, ome_username, ome_password, group
-                    group_id = json_data['value'][0]['Id']
-                    group_payload = {"GroupModel": {
-                        "Name": group_name,
-                        "Description": "",
-                        "MembershipTypeId": 12,
-                        "ParentId": int(group_id)}
-                    }
-                    create_url = 'https://%s/api/GroupService/Actions/GroupService.CreateGroup' % (ome_ip_address)
-                    create_resp = requests.post(create_url, headers=headers,
-                                                verify=False,
-                                                data=json.dumps(group_payload))
-                    if create_resp.status_code == 200:
-                        print("New group created : ID =", create_resp.text)
-                    elif create_resp.status_code == 400:
-                        print("Failed group creation ...See error info below")
-                        print(json.dumps(create_resp.json(), indent=4,
-                                         sort_keys=False))
-            else:
-                print("Unable to retrieve group list from %s" % ome_ip_address)
-        else:
-            print("Unable to create a session with appliance %s" % ome_ip_address)
+
+            group_id = get_group_id_by_name(ome_ip_address, group_name, headers)
+
+            device_ids = []
+            if device_names:
+                for device in device_names:
+                    device_id = get_device_id_by_name(ome_ip_address, device, headers)
+                    if device_id != 0:
+                        device_ids.append(device_id)
+            elif device_tags:
+                device_list = get_device_list(ome_ip_address, headers)
+
+                # Create id - service tag index to avoid O(n) lookups on each search
+                # This is relevant when operating on hundreds of devices
+                id_service_tag_dict = {}
+                for device in device_list:
+                    id_service_tag_dict[device["DeviceServiceTag"]] = device["Id"]
+
+                # Check for each service tag in our index
+                for device_tag in device_tags:
+                    if device_tag in id_service_tag_dict:
+                        device_ids.append(id_service_tag_dict[device_tag])
+                    else:
+                        print("WARNING: Could not find the service tag " + device_tag + ". Skipping.")
+
+            if len(device_ids) > 0:
+                # Add devices to the group
+                payload = {
+                    "GroupId": group_id,
+                    "MemberDeviceIds": device_ids
+                }
+                create_resp = requests.post(group_add_device_url, headers=headers,
+                                            verify=False, data=json.dumps(payload))
+                if create_resp.status_code == 200 or create_resp.status_code == 204:
```

Comment:
The API guide indicates that the return value is a 201 - are you seeing 200 / 204 and not 201? Could you share what version of OME you are using

Reply:
I am receiving 204s here. I'm on Version 3.4.1 (Build 24)

Reply:
I must have looked at an incorrect version of the API guide -it does return a 204 according to the API on page 540